runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java

```diff
@@ -119,6 +119,7 @@
 	private static final Pattern NUMBER_PATTERN = Pattern.compile("([0-9]+)");
 	private static final Pattern LEVEL_UP_PATTERN = Pattern.compile(".*Your ([a-zA-Z]+) (?:level is|are)? now (\\d+)\\.");
 	private static final Pattern BOSSKILL_MESSAGE_PATTERN = Pattern.compile("Your (.+) kill count is: <col=ff0000>(\\d+)</col>.");
+	private static final Pattern VALUABLE_DROP_PATTERN = Pattern.compile("<col=ef1020>(Valuable|Untradeable) drop: ([^<>]+)(?:</col>)?");
```

Comment:
What's the reasoning behind this? Can't this be `<col=ef1020>(Valuable|Untradeable) drop: (.+)(?:</col>)?`

Reply:
I would drop the color specific matching, too

Reply:
> What's the reasoning behind this? Can't this be `<col=ef1020>(Valuable|Untradeable) drop: (.+)(?:</col>)?`

The Valuable drop text will match the </col> text then. Valuable drop has </col>, Untradeable drop doesn't.

Reply:
You should add tests for this, by the way.

Reply:
For this plugin, https://github.com/runelite/runelite/blob/master/runelite-client/src/test/java/net/runelite/client/plugins/screenshot/ScreenshotPluginTest.java

Reply:
Thanks, I looked at some other PRs and didn't see tests, didn't realise not all plugins have tests.

Reply:
This pattern still seems like it is overcomplicated. You do not need to match `[^<>]+` since the pattern matcher will attempt to match the trailing `</col>` before the second group is captured. I would recommend trying to run the tests with the pattern I wrote above.

Reply:
I did try your method, was the first pattern I tried but it doesn't seem to match the trailing part first.

Reply:
https://regexr.com/44bbr ?

Reply:
Ah, I see. Matching to the end of the string and adding a lazy modifier to the `(.+)` group fixes that issue: https://regexr.com/44bcp

Reply:
In any case, I think it's still fine since `<>` are invalid file name characters. Right now there doesn't seem to be any file name checking being done in any of the tests.

Reply:
I suppose that's fair. :+1: